frontend/src/rest/RestClient.js

```diff
@@ -0,0 +1,159 @@
+const BASE_URL = "http://localhost:8080";
+
+class RestClient{
+    constructor(){
+        this.username = "";
+        this.password = "";
+        this.authorization="None";
+    }
+
+    authenticate(username, password){
+        this.username = username;
+        this.password = password;
+        this.authorization = "Basic "+btoa(username + ":" + password);
+    }
+
+    logout(){
+        this.username = "";
+        this.password = "";
+        this.authorization="None";
+    }
+
+    handleListAllQuestions(){
+        return fetch(BASE_URL + "/questions", {
+            method: "GET",
+            headers: {
+                "Authorization": this.authorization
+            }
+        }).then(response => {
+            if(response.status===200)
+                return response.json();
+            return response.text();
+        });
+    }
+
+    handleFilterQuestionsByTitle(content){
+        return fetch(BASE_URL + "/questions/title/"+content.title, {
+            method: "GET",
+            headers: {
+                "Authorization": this.authorization
+            }
+        }).then(response => {
+            if(response.status===200)
+                return response.json();
+            return response.text();
+        });
+    }
+
+    handleFilterQuestionsByTag(content){
+        
+        return fetch(BASE_URL + "/questions/tag/"+content.tagName, {
+            method: "GET",
+            headers: {
+                "Authorization": this.authorization
+            }
+        }).then(response => {
+            if(response.status===200)
+                return response.json();
+            return response.text();
+        });
+    }
+
+    handleCreateQuestion(content) {
+        return fetch(BASE_URL + "/questions", {
+            method: "POST",
+            headers: {
+                "Authorization": this.authorization,
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify({
+                "title": content.title,
+                "text": content.text,
+                "tags": content.tags
+            })
+        }).then(response => {
+            if(response.status===200)
```

Comment:
This is a little hard to read and can lead to (other developers) doing mistakes :) Better use curly brackets and the `else` clause or use the ternary operator: `response.status === 200 ? response.json() : response.text()`.